lib/modules/datasource/npm/get.ts

```diff
@@ -13,11 +14,31 @@ interface PackageSource {
   sourceDirectory?: string;
 }
 
+const SHORT_REPO_REGEX = regEx(/^[A-Za-z0-9_.-]+\/[A-Za-z0-9_.-]+$/);
+
 function getPackageSource(repository: any): PackageSource {
   const res: PackageSource = {};
   if (repository) {
     if (is.nonEmptyString(repository)) {
-      res.sourceUrl = repository;
+      // Put github first because then it matches abc/def too
+      if (SHORT_REPO_REGEX.test(repository.replace(/^github:/, ''))) {
```

Comment:
wouldn't it be better to add the optional prefix to the regex and do a single match?